test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -2211,9 +2211,10 @@ public void run() {
                 // our child) but not our grandchild (i.e. '/bin/sleep'). So
                 // pay attention that the grandchild doesn't run too long to
                 // avoid polluting the process space with useless processes.
-                // Running the grandchild for 60s should be more than enough.
-                final String[] cmd = { "/bin/bash", "-c", "(/bin/sleep 60)" };
-                final String[] cmdkill = { "/bin/bash", "-c", "(/usr/bin/pkill -f \"sleep 60\")" };
+                // Running the grandchild for 59s should be more than enough.
+                // A unique (59s) time is needed to avoid killing other sleep processes.
+                final String[] cmd = { "/bin/bash", "-c", "(/bin/sleep 59)" };
+                final String[] cmdkill = { "/bin/bash", "-c", "(/usr/bin/pkill -f \"sleep 59\")" };
```

Comment:
Maybe future RFE but why do we even need pkill here when we can get the PID of the sleep process we create and kill only that process?

Reply:
I think that's the author's prerogative; the test was written in 2010
to test conditions related to deeply inherited file descriptors.